src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/DefaultHtmlGenerator.cs

```diff
@@ -1096,6 +1095,30 @@ public virtual IHtmlContent GenerateAntiforgery(ViewContext viewContext)
             return currentValues;
         }
 
+        // Public to minimize churn when (2.0?) this is added to IHtmlGenerator.
+        /// <summary>
+        /// Returns a valid HTML 4.01 "id" attribute value for an element with the given <paramref name="fullName"/>.
+        /// </summary>
+        /// <param name="viewContext">A <see cref="ViewContext"/> instance for the current scope.</param>
+        /// <param name="fullName">
+        /// The fully-qualified expression name, ignoring the current model. Also the original HTML element name.
+        /// </param>
+        /// <returns>
+        /// Valid HTML 4.01 "id" attribute value for an element with the given <paramref name="fullName"/>.
+        /// </returns>
+        /// <seealso cref="TagBuilder.CreateSanitizedId"/>
+        public virtual string CreateSanitizedId(ViewContext viewContext, string fullName)
```

Comment:
Normally I'd say that I don't like the idea that this leaks into the public API so prominemently... but almost everything else in this class already requires a `ViewContext`. Are there other APIs here where we should add the `ViewContext`?